File: ui/src/api/cleanup.ts
// 历史数据清理 API
import { post } from './client'

export interface CleanupResult {
  /** 清理的审计日志条数 */
  audit_logs: number
  /** 清理的刷新令牌条数（仅已过期） */
  refresh_tokens: number
}

export interface CleanupPayload {
  /** 清理 days 天以前的数据；不传或 <=0 时用默认 30 天 */
  days?: number
}

// 触发历史数据清理（需 iam:system:cleanup 权限）
export const cleanupHistory = (payload: CleanupPayload = {}) =>
  post<CleanupResult>('/cleanup/history', payload)

File: handler/cleanup.go
package handler

import (
	"net/http"

	"chihqiang/q-iam/logic"

	"github.com/chihqiang/infra-go/httpx"
)

// CleanupHandler 历史数据清理 HTTP 处理器。
type CleanupHandler struct {
	svc *logic.CleanupLogic
}

// NewCleanupHandler 创建历史数据清理处理器。
func NewCleanupHandler(svc *logic.CleanupLogic) *CleanupHandler {
	return &CleanupHandler{svc: svc}
}

// CleanupRequest 清理历史数据请求。
type CleanupRequest struct {
	// Days 清理 days 天以前的数据；<=0 时用默认 30 天。
	Days int `json:"days"`
}

// History 清理历史数据（需 iam:system:cleanup 权限）。
// POST /api/v1/cleanup/history
func (h *CleanupHandler) History(w http.ResponseWriter, r *http.Request) {
	ctx := r.Context()
	var req CleanupRequest
	if err := httpx.MustBindJSON(w, r, &req); err != nil {
		return
	}

	result, err := h.svc.CleanupHistory(ctx, req.Days)
	if err != nil {
		httpx.OkJSONCtx(ctx, w, httpx.NewCodeError(httpx.CodeDefaultError, err.Error()))
		return
	}
	httpx.OkJSONCtx(ctx, w, result)
}
